proptest/src/collection.rs

```diff
@@ -597,90 +600,40 @@ impl<T: ValueTree> ValueTree for VecValueTree<T> {
     type Value = Vec<T::Value>;
 
     fn current(&self) -> Vec<T::Value> {
-        self.elements
-            .iter()
-            .enumerate()
-            .filter(|&(ix, _)| self.included_elements.test(ix))
-            .map(|(_, element)| element.current())
-            .collect()
+        self.current.as_ref().replace(vec![])
     }
 
     fn simplify(&mut self) -> bool {
-        // The overall strategy here is to iteratively delete elements from the
-        // list until we can do so no further, then to shrink each remaining
-        // element in sequence.
-        //
-        // For `complicate()`, we simply undo the last shrink operation, if
-        // there was any.
-        if let Shrink::DeleteElement(ix) = self.shrink {
-            // Can't delete an element if beyond the end of the vec or if it
-            // would put us under the minimum length.
-            if ix >= self.elements.len()
-                || self.included_elements.count() == self.min_size
-            {
-                self.shrink = Shrink::ShrinkElement(0);
-            } else {
-                self.included_elements.clear(ix);
-                self.prev_shrink = Some(self.shrink);
-                self.shrink = Shrink::DeleteElement(ix + 1);
-                return true;
-            }
-        }
-
-        while let Shrink::ShrinkElement(ix) = self.shrink {
-            if ix >= self.elements.len() {
-                // Nothing more we can do
-                return false;
-            }
-
-            if !self.included_elements.test(ix) {
-                // No use shrinking something we're not including.
-                self.shrink = Shrink::ShrinkElement(ix + 1);
-                continue;
-            }
-
-            if !self.elements[ix].simplify() {
-                // Move on to the next element
-                self.shrink = Shrink::ShrinkElement(ix + 1);
-            } else {
-                self.prev_shrink = Some(self.shrink);
-                return true;
-            }
-        }
-
-        panic!("Unexpected shrink state");
+        false
```

Comment:
I understand why `simplify` (or `complicate`) isn't possible for Kani counter-examples, but can you add a short comment?

Reply:
32ce7073475692638acfc594ebb94aa94d82c1d5